catchup/service.go

```diff
@@ -262,11 +262,26 @@ func (s *Service) fetchAndWrite(fetcher Fetcher, r basics.Round, prevFetchComple
 			return false
 		case prevFetchSuccess := <-prevFetchCompleteChan:
 			if prevFetchSuccess {
-				err := s.ledger.AddBlock(*block, *cert)
+				// make sure the ledger wrote enough of the account data to disk, since we don't want the ledger to hold a large amount of data in memory.
+				proto, err := s.ledger.ConsensusParams(r.SubSaturate(1))
+				if err != nil {
+					s.log.Errorf("fetchAndWrite(%d): Unable to determine consensus params for round %d: %v", r, r-1, err)
+					return false
+				}
+				ledgerBacklogRound := r.SubSaturate(basics.Round(proto.MaxBalLookback))
+				select {
+				case <-s.ledger.Wait(ledgerBacklogRound):
```

Comment:
What is the reason for choosing ledgerBacklogRound here? 
I don't mind, but I'm curious.

Reply:
The account updates stores the latest proto.MaxBalLookback rounds in memory. when we get to a higher number, it means that the "distance" between the blocks database and the account database grows. ( i.e. the account database would be lagging behind ).
That "lagging behind" get translated into few problems:
1. the memory consumption grows ( until the data is flushed ).
2. when the account updates try to flush the changes, it would try to flush "too many" changes. This would lead to dbatomic warning messages.
3. If you'll kill the node at any point, it would need to try and recover on startup. To do so, it would need to rescan all the blocks that weren't included in the account database. If you have 1-2 blocks, that's fine. But if you have 1M blocks, it would cause the node to get "hanged" at startup until the issue is resolved.

For these reasons, I think that it would make sense to hold off the writing until we're done flushing the old rounds.. than keep going.